compiler/compile-config/compile-config-2.01-grounding

```diff
@@ -454,65 +413,10 @@ def withRedefinitionsWithJoinsForIdColumns:
-            else
-                rm -f isDenseMultinomial
-                # find distinct weight parameters for factors
-                # considering all the exact combintations of category values present in the factors
-                # assuming they'll be sparse
-                deepdive create table \(.weightsTable | @sh) as \(
-                    { SELECT:
-                        # weight parameters
-                        [ ( .weight_.params[] | { column: . } )
-                        # include category columns for every categorical variable
-                        , ( .function_.variables[]
-                        | .columnId as $columnIdForThisVar
-                        | .schema.variablesCategoryColumns[]
-                        | { alias: "\($columnIdForThisVar)_\(.)", table: "v.\($columnIdForThisVar)", column: . } )
-                        # weight attributes
-                        , { alias: "isfixed"  , expr: .weight_.is_fixed      }
-                        , { alias: "initvalue", expr: .weight_.init_value    }
-                        , { alias: "wid"      , expr: "CAST(NULL AS BIGINT)" } # to be assigned later by the assign_weight_id process
-                        ]
-                    , DISTINCT: true  # XXX this is inevitable
-                    # weights are exploded for the factor's variables' category values
-                    , FROM: [ { alias: "f", table: .factorsTable } ]
-                    # FIXME joins below can go away if DDlog could project them as well in the .input_query
-                    # or we could inject those pushed-down projections if it at least compiled into a sql.jq object
-# XXX this join prevents us from defining a partial weight distribution
-                    , JOIN:
-                        [ ( .function_.variables[]
-                        | .columnId as $columnIdForThisVar
-                        | { INNER: { alias: "i.\($columnIdForThisVar)", table: .schema.variablesIdsTable }
-                          , ON: { eq: [ { table: "f"                          , column: $columnIdForThisVar }
-                                      , { table: "i.\($columnIdForThisVar)", column: deepdiveVariableIdColumn } ] }
-                          }
-                        , { INNER: { alias: "v.\($columnIdForThisVar)", table: .schema.variablesTable }
-                          , ON: [ ( .schema.variablesKeyColumns[]
-                                | { eq: [ { table: "v.\($columnIdForThisVar)", column: . }
-                                        , { table: "i.\($columnIdForThisVar)", column: . } ] } ) ] }
-                        ) ]
-                    } | asSql | asPrettySqlArg)
-            fi
-            " else # factors over boolean variables
             "
             # find distinct weights for the factors into a separate table
             deepdive create table \(.weightsTable | @sh) as \($weightsMaterializeSqlArg)
-            " end)
+            ")
```

Comment:
For readability let's inline `$weightsMaterializeSqlArg` since it's no longer repeated.